### DISCUSSION TITLE
Speaker identification?

### DISCUSSION THREAD

**Discussion by thigger** [Pikurrot/whisper-gui#27] [Ideas]
Is there any way/plan to support speaker identification in this? The transcription is great but for my use-case I need to know who said what; and it looks like there is a way to support identification in WhisperX. Thanks!

**Comment by Pikurrot** [comment]
I was planning to do speaker diarization in the long term. Should be possible with whisperx as there is this defined class [here](https://github.com/m-bain/whisperX/blob/9e3a9e0e38fcec1304e1784381059a0e2c670be5/whisperx/diarize.py). I'm not currently aware whether it's possible with a custom whisper model.
I will try to do that anytime soon, but feel free to contribute.

**Reply by thigger** [reply]
Thanks. I'll have a look to see if I can help try something. I think I've seen elsewhere people complaining that the diarisation model is gated but I guess having it as an option (with token required) lets people decide.
